File: official/docs/python/v7/smartrate/retrieve-estimated-delivery-date.py
import easypost

easypost.api_key = "EASYPOST_API_KEY"

shipment = easypost.Shipment.retrieve("shp_...")

estimated_delivery_dates = shipment.retrieve_estimated_delivery_date(planned_ship_date="YYYY-MM-DD")

print(estimated_delivery_dates)

File: official/docs/golang/v3/smartrate/retrieve-estimated-delivery-date.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v3"
)

func main() {
	client := easypost.New("EASYPOST_API_KEY")

	shipment, _ := client.GetShipment("shp_...")
	estimatedDeliveryDates, _ := client.GetShipmentEstimatedDeliveryDate(shipment.ID, "YYYY-MM-DD")

	fmt.Println(estimatedDeliveryDates)
}
